tools/clang/test/HLSLFileCheck/validation/WaveSizeAttrAllowed.hlsl

```diff
@@ -0,0 +1,15 @@
+// RUN: %dxc -E main -T lib_6_7 %s | FileCheck %s
+
+// This test used to fail because it didn't take into account
+// the possibility of the compute shader being a library target.
+// Now, this test is an example of what should be allowed
+// and no errors are expected.
+
+// CHECK: define void @main()
+
+[Shader("compute")]
+[WaveSize(8)]
+[numthreads(2,2,2)]
```

Comment:
FYI, by just removing the numthreads line on the commit that you reviewed, this is emitted:
```
error: validation errors
D:\DXC\tools\clang\test\HLSLFileCheck\validation\WaveSizeAttrAllowed.hlsl:12: error: Declared Thread Group X size 3575194952 outside valid range [1..1024].
D:\DXC\tools\clang\test\HLSLFileCheck\validation\WaveSizeAttrAllowed.hlsl:12: error: Declared Thread Group Z size 3575196168 outside valid range [1..64].
D:\DXC\tools\clang\test\HLSLFileCheck\validation\WaveSizeAttrAllowed.hlsl:12: error: Declared Thread Group Count 3743912256 (X*Y*Z) is beyond the valid maximum of 1024.
Validation failed.
```
That is why I had to add numthreads.

Reply:
As for the case where a shader just has the WaveSize and shader("node") attributes, should the failure reason only be because of an invalid attribute combination?

Reply:
That's not a great error, but it's a separate issue. I think wavesize and numthreads should fail with this error for the sake of including numthreads:
`          DiagnosticsEngine::Error, "attribute numthreads only valid for CS/MS/AS.");`

There should be a similar error for wavesize alone in a lib shader:
       ` DiagnosticsEngine::Error,  "attribute WaveSize only valid for CS.");`